runtime/tests/interpreter/runtimetype_test.go

```diff
@@ -0,0 +1,728 @@
+			Type: interpreter.ConstantSizedStaticType{
+				Type: interpreter.CompositeStaticType{
+					Location:            utils.TestLocation,
+					QualifiedIdentifier: "R",
+					TypeID:              "S.test.R",
+				},
+				Size: int64(400),
+			},
+		},
+		inter.Globals["c"].GetValue(),
+	)
+
+	assert.Equal(t,
+		interpreter.TypeValue{
+			Type: interpreter.ConstantSizedStaticType{
+				Type: interpreter.ConstantSizedStaticType{
+					Type: interpreter.PrimitiveStaticTypeString,
+					Size: int64(10),
+				},
+				Size: int64(6),
+			},
+		},
+		inter.Globals["d"].GetValue(),
+	)
+
+	assert.Equal(t,
+		inter.Globals["a"].GetValue(),
+		inter.Globals["e"].GetValue(),
+	)
+}
+
+func TestInterpretDictionaryType(t *testing.T) {
+
+	t.Parallel()
+
+	inter := parseCheckAndInterpret(t, `
+      let a = DictionaryType(Type<String>(), Type<Int>())!
+      let b = DictionaryType(Type<Int>(), Type<String>())!
+
+      resource R {}
+      let c = DictionaryType(Type<Int>(), Type<@R>())!
+      let d = DictionaryType(Type<Bool>(), a)!
+
+      let e = Type<{String: Int}>()!
+      
+      let f = DictionaryType(Type<[Bool]>(), Type<Int>())
+    `)
+
+	assert.Equal(t,
+		interpreter.TypeValue{
+			Type: interpreter.DictionaryStaticType{
+				KeyType:   interpreter.PrimitiveStaticTypeString,
+				ValueType: interpreter.PrimitiveStaticTypeInt,
+			},
+		},
+		inter.Globals["a"].GetValue(),
+	)
+
+	assert.Equal(t,
+		interpreter.TypeValue{
+			Type: interpreter.DictionaryStaticType{
+				KeyType:   interpreter.PrimitiveStaticTypeInt,
+				ValueType: interpreter.PrimitiveStaticTypeString,
+			},
+		},
+		inter.Globals["b"].GetValue(),
+	)
+
+	assert.Equal(t,
+		interpreter.TypeValue{
+			Type: interpreter.DictionaryStaticType{
+				ValueType: interpreter.CompositeStaticType{
+					Location:            utils.TestLocation,
+					QualifiedIdentifier: "R",
+					TypeID:              "S.test.R",
+				},
+				KeyType: interpreter.PrimitiveStaticTypeInt,
+			},
+		},
+		inter.Globals["c"].GetValue(),
+	)
+
+	assert.Equal(t,
+		interpreter.TypeValue{
+			Type: interpreter.DictionaryStaticType{
+				ValueType: interpreter.DictionaryStaticType{
+					KeyType:   interpreter.PrimitiveStaticTypeString,
+					ValueType: interpreter.PrimitiveStaticTypeInt,
+				},
+				KeyType: interpreter.PrimitiveStaticTypeBool,
+			},
+		},
+		inter.Globals["d"].GetValue(),
+	)
+
+	assert.Equal(t,
+		inter.Globals["a"].GetValue(),
+		inter.Globals["e"].GetValue(),
+	)
+
+	assert.Equal(t,
+		interpreter.NilValue{},
+		inter.Globals["f"].GetValue(),
+	)
+}
+
+func TestInterpretCompositeType(t *testing.T) {
+
+	t.Parallel()
+
+	inter := parseCheckAndInterpret(t, `
+      resource R {}
+      struct S {}
+      struct interface B {}
+
+      let a = CompositeType("S.test.R")!
+      let b = CompositeType("S.test.S")!
+      let c = CompositeType("S.test.A")
+      let d = CompositeType("S.test.B")
+
+      let e = Type<@R>()
+
+      enum F: UInt8 {}
+      let f = CompositeType("S.test.F")!
```

Comment:
Shall we add some tests to cover native/built-in composite types (e.g: `PublicKeyType`, `HashAlgorithmType`) as well?